src/vm/arm64/asmhelpers.asm

```diff
@@ -1326,5 +1329,62 @@ Fail
     LEAF_END
 #endif
 
+;
+; JIT Static access helpers when coreclr host specifies single appdomain flag 
+;
+
+; ------------------------------------------------------------------
+; void* JIT_GetSharedNonGCStaticBase(SIZE_T moduleDomainID, DWORD dwClassDomainID)
+
+    LEAF_ENTRY JIT_GetSharedNonGCStaticBase_SingleAppDomain
+    ; If class is not initialized, bail to C++ helper
+    add x2, x0, #DomainLocalModule__m_pDataBlob
+    ldrb w2, [x2, w1]
```

Comment:
I am using ARM DDI 0487A.j Rev J of the ARM Architecture Reference Manual
ARMv8, for ARMv8-A architecture profile.

The form is invalid because 
```
ldrb w2, [x2, w1]
```
Requires the a non-optional specification of how to convert the lower 32 bits of w1 into 64 bits.  Either
```
ldrb w2, [x2, w1, UXTW]
ldrb w2, [x2, w1, SWTW]
```
Since r1 is already a 64 bit register, it seemed strange to arbitrarily truncate to 32 bits then extend.  So I assumed you intended
```
ldrb w2, [x2, x1]
```
a synonym of 
```
ldrb w2, [x2, x1, LSL #0]
```